src/value.rs

```diff
@@ -91,6 +93,13 @@ pub fn lua_to_rbxvalue(ty: RbxValueType, value: LuaValue<'_>) -> LuaResult<RbxVa
             value: value as i64,
         }),
 
+        (RbxValueType::Color3, LuaValue::UserData(ref user_data))
+            if user_data.is::<Color3Value>() =>
+        {
+            let color = &*user_data.borrow::<Color3Value>().unwrap();
+            Ok(color.into())
+        }
+
```

Comment:
Since `user_data.borrow::<T>()` will tell us if we couldn't do that borrow, I think we should drop the `if` guard on the pattern and do that check internally.

We might need to finagle it a bit to get it to compile, but something like this should work:

```rust
(RbxValueType::Color3, LuaValue::UserData(ref user_data)) => {
    let color = user_data.borrow::<Color3Value>()?;
    Ok(color.into())
}
```

Note that the error type returned by `borrow` is `rlua::Error` I think, which means we can use Rust's `?` operator!

This means we'll be duplicating the error return case, but I think that's okay. If it ends up being a problem, I like [hoisting the error type into a closure](https://github.com/rojo-rbx/rbx-dom/blob/412d996e55ceb261eaf75f541cb7bf41f2778eeb/rbx_binary/src/serializer.rs#L459-L470).